### DISCUSSION TITLE
fp-ts and effect-ts

### DISCUSSION THREAD

**Discussion by shinzui** [gcanti/fp-ts#1852] [General]
Hi,

I am confused about the [announcement](https://dev.to/effect-ts/a-bright-future-for-effect-455m) of `fp-ts` joining the `effect-ts` ecosystem. Can someone please clarify the implication of the announcement on `fp-ts`, `io-ts`, and `monocle-ts`?

**Comment by gcanti** [comment]
All new developments will happen under the effect org, roughly speaking there are the following correspondencies:

- fp-ts = effect/data + effect/io
- io-ts = effect/schema
- monocle-ts = fp-ts/optic (not yet accepted into the effect org)

When possible, some of the features designed and implemented in the effect/* packages will be backported to fp-ts in order to

- make easier switching to effect
- make fp-ts better for users who don't want or can't switch to effect

**Reply by p3tr2005** [reply]
thanks, dude. now i don't have to switch into effect cause i like the fp-ts so much.

**Comment by shinzui** [comment]
@gcanti Thank you for answering. I'm one of the users who don't want to switch to effect, so it's reassuring that `fp-ts` will not be only in maintenance mode.

**Comment by nandin-borjigin** [comment]
There is one more package `@effect/typeclass` which has the typeclasses using v3 HKT.

**Comment by albertodiazdorado** [comment]
@gcanti If we are using `fp-ts`, should we migrate our projects to `effect-ts`? Is there any timeline (or deadline) for the migration?

**Reply by gcanti** [reply]
"should"? not at all, it's only worth migrating to effect-ts if the benefits it offers outweigh the costs of migration

**Reply by albertodiazdorado** [reply]
Oh perfect, that's great!

I had (mis)understood that `fp-ts` was on the way to being discontinued (as in "_We backport bugfxix in Node20 to Node16, but eventually Node16 will no longer be supported_" ). Thanks for the clarification :)

**Comment by noncom** [comment]
I totally don't understand this. Could someone please actually explain?

Looking at both fp-ts and Effect all I see is that these libraries have quite different functionalities with just _some **minor**_ intersection:
- Effect 
  - All about application-level composition, 
  - Task composition and scheduling, including IO, etc
  - Catching errors in FP-like way. This concludes the whole FP part. It's like 5% of what fp-ts has.
  - Schema verification
  - Other practical things, Effect is simply bigger, having a lot of stuff for writing actual application architecture. It's going to be all over the place.
- fp-ts
  - A ton of FP-related type-theoretic concepts like HKT, Lattices, Group, Category, Comonad, etc
  - Some provisional task management, just to kind of apply some FP to it, but apparently just not so advanced as in Effect
  - Focuses on math-y things, without any attempt to influence how you architecture your app.

The fact that they both have the `Either` type, some means to wrap errors and tasks, doesn't make them two versions of the same library. And it is impossible to comprehend how Effect is a better version of fp-ts.

What am I missing? How do we arrive at the conclusion that Effect is fp-ts v3?
